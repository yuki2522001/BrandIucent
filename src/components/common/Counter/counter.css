.counter {
  position: relative;
  z-index: 15;
  width: 205px;
}

.counter__input {
  border: 1px solid var(--border-primary);
  border-radius: 24.5px;
  text-align: center;
  padding: 14px;
}

.counter__minus {
  position: absolute;
  z-index: 20;
  top: 33%;
  left: 22px;
}

.counter__plus {
  position: absolute;
  z-index: 20;
  top: 35%;
  right: 25px;
}