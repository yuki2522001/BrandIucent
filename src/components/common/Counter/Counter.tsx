import React, { memo, useState } from "react";
import Icon from "../Icon/Icon";
import "./counter.css";

interface CounterProps {
  counter?: number;
}

const Counter: React.FC<CounterProps> = memo(({ counter }) => {
  const [count, setCount] = useState<number>(counter!);

  const handleIncrease = () => {
    setCount(count + 1);
  };

  const handleDecrease = () => {
    setCount(count - 1);
  };

  return (
    <div data-testid="counter" className="counter">
      <div className="counter__minus">
        <Icon onClick={handleDecrease} iconName="minus" />
      </div>
      <input className="counter__input" value={count} min={0} />
      <div className="counter__plus">
        <Icon onClick={handleIncrease} iconName="plus" />
      </div>
    </div>
  );
});

export default Counter;
