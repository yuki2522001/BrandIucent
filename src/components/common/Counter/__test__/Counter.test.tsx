import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Counter from "../Counter";

describe("Counter component", () => {
  test("should render Counter component", () => {
    const { getByTestId } = render(<Counter counter={0} />);
    expect(getByTestId("counter")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<Counter counter={0} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
