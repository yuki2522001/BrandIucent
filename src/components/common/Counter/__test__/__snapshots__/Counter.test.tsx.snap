// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="counter"
    data-testid="counter"
  >
    <div
      class="counter__minus"
    >
      <i
        class="icon-minus"
        data-testid="icon"
      />
    </div>
    <input
      class="counter__input"
      min="0"
      value="0"
    />
    <div
      class="counter__plus"
    >
      <i
        class="icon-plus"
        data-testid="icon"
      />
    </div>
  </div>
</DocumentFragment>
`;
